tests/framework/hooks/test_hook_log_parameters.py

```diff
@@ -74,6 +75,60 @@ def dummy_catalog():
     return catalog
 
 
+@pytest.mark.parametrize(
+    "param_name,expected_name",
+    [
+        ("valid_param", "valid_param"),
+        ("valid:param", "valid:param"),
+        ("valid-param", "valid-param"),
+        ("invalid/param", "invalid/param"),
+        ("invalid.param", "invalid.param"),
+        ("[invalid]$param", "_invalid__param"),
+    ],
+)
+def test_parameter_name_sanitization(
+    kedro_project, dummy_run_params, param_name, expected_name
+):
+    _write_yaml(
+        kedro_project / "conf" / "local" / "mlflow.yml",
+        dict(
+            tracking=dict(params=dict(long_params_strategy="fail")),
+        ),
+    )
+
+    mlflow_tracking_uri = (kedro_project / "mlruns").as_uri()
+    mlflow.set_tracking_uri(mlflow_tracking_uri)
+
+    node_inputs = {f"params:{param_name}": "test_value"}
+
+    bootstrap_project(kedro_project)
+    with KedroSession.create(
+        project_path=kedro_project,
+    ) as session:
+        context = session.load_context()
```

Comment:
Just a nitpick, but can we just ignore all this? Since you invoke the hook manually, I don't think it's necessary creating a whole session and contexte, unless I miss something?

Reply:
My bad, this is needed to call the after_context_created hook, which format and stores the entire configuration.